agent/src/datafile.rs

```diff
@@ -773,7 +762,10 @@ impl DataFile {
         let region = region.unwrap();
 
         match region.state {
-            State::Requested | State::Destroyed | State::Tombstoned => {
```

Comment:
I don't think Nexus is doing the right thing if it's asking for snapshots from a failed region - where did the call come from?

Reply:
As part of the region delete, it first checks for snapshots,   `agent/src/server.rs`:

```
#[endpoint {                                                                      
    method = DELETE,                                                              
    path = "/crucible/0/regions/{id}",                                            
}]                                                                                
async fn region_delete(                                                           
    rc: RequestContext<Arc<DataFile>>,                                            
    path: TypedPath<RegionPath>,                                                  
) -> SResult<HttpResponseDeleted, HttpError> {                                    
    let p = path.into_inner();                                                    
                                                                                  
    // Cannot delete a region that's backed by a ZFS dataset if there are         
    // snapshots.                                                                 
                                                                                      let snapshots = match rc.context().get_snapshots_for_region(&p.id) {          
        Ok(results) => results,                                                   
        Err(e) => {                                                               
            return Err(HttpError::for_internal_error(e.to_string()));             
        }                                                                         
    };
```

Without that, the delete is refused because we get an error back from the get_snapshots call.